src/mono/mono/mini/interp/transform.c

```diff
@@ -5084,6 +5087,15 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 			WRITE64_INS (td->last_ins, 0, &counter);
 		}
 
+		if (generate_enc_seq_points_without_debug_info)
+		{
+			if ((*td->ip == CEE_NOP) ||
+				(*td->ip == CEE_CALLVIRT) ||
+				(*td->ip == CEE_CALLI) ||
+				(*td->ip == CEE_CALL) ||
+				(GPTRDIFF_TO_INT (td->sp - td->stack) == 0))
```

Comment:
Can this be a helper function.  Also add a comment that this is matching CoreCLR's algorithm for adding sequence points